smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/UnionGenerator.java

```diff
@@ -203,5 +224,55 @@ private void writeVisitorFunction() {
         }
         writer.write("return visitor._(value.$$unknown[0], value.$$unknown[1]);");
         writer.dedent().write("}");
+        writer.write("");
+    }
+
+    private void writeFilterSensitiveLog() {
+        String objectParam = "obj";
+        writer.openBlock("export const filterSensitiveLog = ($L: $L): any => {", "}",
+            objectParam, symbol.getName(),
+            () -> {
+                for (MemberShape member : shape.getAllMembers().values()) {
+                    Shape memberTarget = model.expectShape(member.getTarget());
+                    String memberName = symbolProvider.toMemberName(member);
+
+                    if (member.getMemberTrait(model, SensitiveTrait.class).isPresent()) {
```

Comment:
A lot of this code, specifically the branching, looks similar to that of writing this function for structures (`StructuredMemberWriter.writeFilterSensitiveLog`.) Can it be consolidated cleanly at all?